### ISSUE TITLE
database secrets engine create_role broken

### ISSUE THREAD

**Issue by drewmullen** [hvac/hvac#458]
`print(client.secrets.database.create_role(name='tester', db_name='test', creation_statements=[]))
`

error: `Exception: empty database name attribute`

```
vault list database/config
Keys
----
test
```

```
Exception has occurred: SystemExit
1
  File "/Users/dmullen/venv/lib/python3.7/site-packages/ansible/module_utils/basic.py", line 2370, in fail_json
    sys.exit(1)
  File "/Users/dmullen/venv/lib/python3.7/site-packages/ansible/modules/hashivault/hashivault_db_secret_engine_role.py", line 117, in main
    module.fail_json(**result)
  File "/Users/dmullen/venv/lib/python3.7/site-packages/ansible/modules/hashivault/hashivault_db_secret_engine_role.py", line 192, in <module>
    main()
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
```

**Comment by drewmullen** [comment]
`curl -X POST -H 'X-Vault-Token: <redacted>' --data @db-role-payload.json  -k $VAULT_ADDR/v1/database/roles/tester`

```
$ cat db-role-payload.json
{
      "creation_statements": [],
      "revocation_statements": [],
      "rollback_statements": "",
      "db_name": "test"
}
```

```
vault read database/roles/tester
Key . Value
----   -----
creation_statements ...
...
```

**Comment by drewmullen** [comment]
something is definitely up with the module... investigate

@Dudesons any ideas?

**Comment by drewmullen** [comment]
so it appears as though `requests` lib sends args at the end of the url, not sure if thats normal or not tbh (see below). @jeffwecan any idea if thats normal?

by the time it gets to `requests` lib heres some of the 

prep: `'/v1/database/roles/tester?db_name=test&default_ttl=0&max_ttl=0'`
send_kwargs: `{'allow_redirects': True, 'cert': None, 'proxies': OrderedDict([]), 'stream': False, 'timeout': 30, 'verify': False}`

and the resp from vault is:
rc 400
b'{"errors":["empty database name attribute"]}\n'

**Comment by drewmullen** [comment]
```
curl -v -i -X POST -H 'X-Vault-Token: <redacted>' -k 'https://vault.example.com:8200/v1/database/roles/tester2?db_name=test&default_ttl=0&max_ttl=0'
```
error: `{"errors":["empty database name attribute"]}`

😮

**Comment by drewmullen** [comment]
hmm, this is working:
```
r = requests.post("https://vault.example.com:8200/v1/database/roles/tester3", headers=headers, json=payload, verify=False)
print(r)
<Response [204]>
```

**Comment by drewmullen** [comment]
so it seems like requests works when you pass the args via `json` param but **NOT** if you use the **params** parameter

@jeffwecan would love your opinion on this... is this a hashi bug in their http api or a problem with hvac design?

**Comment by drewmullen** [comment]
i think maybe this should be `json`

https://github.com/hvac/hvac/blob/e8ae13ac69611a97a3499e2eaa99351b5d5b7556/hvac/api/secrets_engines/database.py#L156

**Comment by drewmullen** [comment]
yeah thats it. 

also, this doesnt return a json object
https://github.com/hvac/hvac/blob/e8ae13ac69611a97a3499e2eaa99351b5d5b7556/hvac/api/secrets_engines/database.py#L157

ill push a fix now

**Comment by drewmullen** [comment]
this also doesnt return a json object

https://github.com/hvac/hvac/blob/e8ae13ac69611a97a3499e2eaa99351b5d5b7556/hvac/api/secrets_engines/database.py#L197

**Comment by drewmullen** [comment]
this also doesnt return a json object

https://github.com/hvac/hvac/blob/e8ae13ac69611a97a3499e2eaa99351b5d5b7556/hvac/api/secrets_engines/database.py#L117

**Comment by drewmullen** [comment]
im going to also guess that `rotate_root_credentials()` does not return a json object. but i cannot test this right now

**Comment by Dudesons** [comment]
Hi @drewmullen, yes sorry I begin on some get request and forgot to update request to `json` params.
Do you want to do the pull request or I can fix that and add documentation (#453)

**Comment by drewmullen** [comment]
@Dudesons all fixed :) https://github.com/hvac/hvac/pull/457
